microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/FakeJavaProject.java

```diff
@@ -58,13 +61,15 @@ public FakeJavaProject(IJavaProject rootProject, IClasspathEntry[] entries) thro
 		if (MicroProfilePropertiesScope.isOnlySources(scopes)) {
 			return new IJavaElement[] { rootProject };
 		}
+		List<IJavaElement> elements = new ArrayList<>();
+		elements.add(rootProject);
 		IPackageFragmentRoot[] roots = super.getPackageFragmentRoots();
-		IJavaElement[] elements = new IJavaElement[1 + roots.length];
-		elements[0] = rootProject;
-		for (int i = 0; i < roots.length; i++) {
-			elements[i + 1] = roots[i];
+		for (IPackageFragmentRoot root : roots) {
+			if (searchClasspathEntries.contains(root.getResolvedClasspathEntry())) {

```

Comment:
Could you please explain what this check is for? I'm having trouble understanding this statement

Reply:
Filtering classpath entries to get the ones we need? It looks like `QuarkusConfigRootProvider.contributeToClasspath()` adds entries to `searchClasspathEntries`. 

Should `searchClasspathEntries` only contain deployment artifacts?

Reply:
I remove this code. Don't need to do that.